packages/synapse-react-client/src/utils/jsonschema/getType.ts

```diff
@@ -1,21 +1,43 @@
 import { getOnlyNonNullOneOfOption } from '@/utils/jsonschema/getOnlyNonNullOneOfOption'
-import { JSONSchema7Type } from 'json-schema'
+import { JSONSchema7 } from 'json-schema'
 import isArray from 'lodash-es/isArray'
 
-export function getType(
-  jsonSchema: Record<string, unknown>,
-): JSONSchema7Type | undefined {
+export interface TypeInfo {
```

Comment:
Refactored this quite a bit to handle array schemas in a somewhat consistent way with json. I'm open to creating a new function, like getTypeDetails